worker/uniter/remotestate/watcher.go

```diff
@@ -230,29 +233,32 @@ func (w *RemoteStateWatcher) loop(unitTag names.UnitTag) (err error) {
 		seenStorageChange bool
 		storageChanges    watcher.StringsChannel
 
-		seenApplicationChange bool
-		applicationChanges    watcher.NotifyChannel
+		seenApplicationChange *bool
 	)
-
-	if w.modelType == model.IAAS {
-		storagew, err := w.unit.WatchStorage()
+	if w.applicationChannel == nil {
+		// This is in IAAS model so we need to watch state for application
```

Comment:
It seems weird that you have split up the `if IAAS model`. 

Is there a Validate on the WatcherConfig struct? If not, there probably should be to assert that a CAAS model has an applicationChannel. Then keep this defined in terms of model type rather than splitting based on applicationChannel, then below, the model type.

Reply:
Added validate()